File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_154/typescript/solution.ts
function cycpattern_check(a: string, b: string): boolean {
  const doubled = b + b;
  for (let i = 0; i < b.length; i++) {
    if (a.includes(doubled.substring(i, i + b.length))) {
      return true;
    }
  }
  return false;
}

File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_154/csharp/solution.cs
public static bool CycpatternCheck(string a, string b) {
    for (int i = 0; i < b.Length; i++) {
        string rotation = b[i..] + b[..i];
        if (a.Contains(rotation)) {
            return true;
        }
    }
    return false;
}
